app/api/main/controller/company_controller.py

```diff
@@ -56,28 +60,108 @@ async def full_evaluation(search_request: SearchRequest) -> EvaluatedCompanyDTO:
+    try:
+        for company_request in request:
+            try:
+                # Find the company by ID
+                company = await evaluated_companies_collection.find_one({"_id": ObjectId(company_request.id)})
+                if not company:
+                    failed_toggles.append({"id": company_request.id, "reason": "Company not found"})
+                    continue
+
+                # Get the current compliance value
+                current_compliance = company.get("compliance", False)
+
+                # Toggle the compliance value
+                new_compliance_status = await CompanyService.toggle_compliance(company_request.id, current_compliance)
+                success_toggles.append({
+                    "id": company_request.id,
+                    "name": company["name"],
+                    "compliance": new_compliance_status
+                })
+            except Exception as e:
+                failed_toggles.append({"id": company_request.id, "reason": str(e)})
+
+        # Construct a response message
+        if len(success_toggles) == 1:
+            message = f"Compliance for company '{success_toggles[0]['name']}' with ID '{success_toggles[0]['id']}' updated successfully."
+        elif len(success_toggles) > 1:
+            companies_list = ", ".join(
+                [f"{company['name']} with ID {company['id']}" for company in success_toggles[:-1]]
+            )
+            last_company = f"{success_toggles[-1]['name']} with ID {success_toggles[-1]['id']}"
+            message = f"Compliance for companies {companies_list}, and {last_company} updated successfully."
+        else:
+            message = "No companies were updated."
+
+        # Return the response
+        return {
+            "message": message,
+            "success": success_toggles,
+            "failed": failed_toggles
+        }
+
+    except Exception as e:
         raise HTTPException(
-            status_code=404, detail=f"Company '{request.company_name}' not found."
+            status_code=500,
+            detail=f"An error occurred: {str(e)}"
         )
 
-    # Get the current compliance value
-    current_compliance = company.get("compliance", False)
+
+@router.delete("/delete-companies")
+async def delete_companies(request: List[CompanyRequest]):
+    company_ids = [ObjectId(company_request.id) for company_request in request]
+    failed_deletions = []
+    success_deletions = []
 
     try:
-        await CompanyService.toggle_compliance(request.company_name, current_compliance)
+        # Fetch all companies to validate existence
+        for company_id in company_ids:
+            company = await CompanyService.get_evaluated_company(company_id)
+            if not company:
+                failed_deletions.append(f"Company with ID '{company_id}' not found.")
+                continue
+            # Append a JSON object
+            success_deletions.append({"name": company["name"], "id": str(company_id)})
+
```

Comment:
Need error handling for when company was found but failed due to another reason
```suggestion
except Exception as e:
                failed_companies.append({"id": company_request.id, "reason": str(e)})
```

Reply:
Already done, mind checking again?